draft-ietf-quic-transport.md

```diff
@@ -908,10 +908,13 @@ final size is the number of bytes sent.  More generally, this is one higher
 than the offset of the byte with the largest offset sent on the stream, or zero
 if no bytes were sent.
 
-For a stream that is reset, the final size is carried explicitly in a
-RESET_STREAM frame.  Otherwise, the final size is the offset plus the length of
-a STREAM frame marked with a FIN flag, or 0 in the case of incoming
-unidirectional streams.
+The final size of a stream is always signaled to the recipient.  The final size
+is the sum of the Offset and Length fields of a STREAM frame with a FIN flag,
+noting that these fields might be implicit.  Alternatively, the Final Size field
+of a RESET_STREAM frame carries this value.  This ensures that all ways that a
+stream can be closed result in the number of bytes on the stream being reliably
```

Comment:
I don't think this is quite correct, since there's not an implicit fin in the case of a stream where the Fin was never sent and the stream was reset.

```suggestion
The final size of a stream is always signaled to the recipient.  For streams that
are completed, the final size is the sum of the Offset and Length fields of a
STREAM frame with a FIN flag.  For a stream that is reset, the Final Size field
of a RESET_STREAM frame carries the final size.  This ensures that all ways that a
stream can be closed result in the number of bytes on the stream being reliably
```

Reply:
The implicit refers to the Length field, which is implicit when omitted.  Some tweaking might be required for that.

Reply:
No need to refer to the wire encoding of offset and length, right?
My suggestion: "... is the sum of the offset and the length of a STREAM frame with a FIN flag"

Reply:
Both the Offset and Length fields can be implicit.  As discussed above, it would be unusual (at least with HTTP/3 for both to be implicit, but it's certainly allowed.  I think if the fields specifically get mentioned, the aside that they might be implicit needs to remain.  If we switch to pointing to the values, we could drop that, but it might be confusing to some people.

FIN isn't a field, and isn't contemplated by the aside about implicit fields.